### PR TITLE
Don't use else after return

### PR DESCRIPTION
**Pull request by endJunction**
clang-tidy's automatic changes. Same code with less code ;)

### CODE REVIEW

**Comment by chleh** [comment]
I'm fine with it, but I didn't look at it in detail. :+1:

**Comment by chleh** [comment]
Btw: Do you have a plan which clang-tidy things you are going to do in total?

**Comment by endJunction** [comment]
`clang-tidy -list-checks -checks='*'`
Already excluded are:
 - llvm-header-guard,
 - google-readability-todo,
 - readability-avoid-const-params-in-decls

But we could take them in, if desired.

File: Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp [hunk 1]
```diff
@@ -101,22 +101,18 @@ QVariant VtkAlgorithmProperties::GetUserProperty(QString name) const
 {
     if (this->_algorithmUserProperties->contains(name))
         return this->_algorithmUserProperties->value(name);
-    else
-    {
+
         ERR("Not a valid property: %s", name.toStdString().c_str());
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp]: Indentation of line 105 and 106 seems to be wrong.

File: Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp [hunk 2]
```diff
@@ -101,22 +101,18 @@ QVariant VtkAlgorithmProperties::GetUserProperty(QString name) const
...
 
 QList<QVariant> VtkAlgorithmProperties::GetUserVectorProperty(QString name) const
 {
     if (this->_algorithmUserVectorProperties->contains(name))
         return this->_algorithmUserVectorProperties->value(name);
-    else
-    {
+
         ERR("Not a valid property: %s", name.toStdString().c_str());
         return QList<QVariant>();
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkAlgorithmProperties.cpp]: Indentation of line 114 and 115 seems to be wrong.

File: Applications/DataExplorer/VtkVis/VtkFilterFactory.cpp [hunk 3]
```diff
@@ -111,9 +111,9 @@ VtkCompositeFilter* VtkFilterFactory::CreateCompositeFilter( QString type,
 {
     if (type.compare(QString("VtkCompositeImageToCylindersFilter")) == 0)
         return new VtkCompositeImageToCylindersFilter(inputAlgorithm);
-    else if (type.compare(QString("VtkCompositePointToGlyphFilter")) == 0)
+    if (type.compare(QString("VtkCompositePointToGlyphFilter")) == 0)
         return new VtkCompositePointToGlyphFilter(inputAlgorithm);
-    else if (type.compare(QString("VtkCompositeLineToTubeFilter")) == 0)
+    if (type.compare(QString("VtkCompositeLineToTubeFilter")) == 0)
         return new VtkCompositeLineToTubeFilter(inputAlgorithm);
     else if (type.compare(QString("VtkCompositeColormapToImageFilter")) == 0)
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkVis/VtkFilterFactory.cpp]: There are some more occurences of `else if` that could be fixed.

File: GeoLib/GEOObjects.cpp [hunk 4]
```diff
@@ -235,16 +235,14 @@ bool GEOObjects::appendSurfaceVec(const std::vector<Surface*>& surfaces,
...
-    else
-    {
+
         // the copy is needed because addSurfaceVec is passing it to SurfaceVec
         // ctor, which needs write access to the surface vector.
         auto sfc = std::make_unique<std::vector<GeoLib::Surface*>>();
         for (auto surface : surfaces)
             sfc->push_back(surface);
         addSurfaceVec(std::move(sfc), name);
         return false;
```
Comment (TomFischer) [inline GeoLib/GEOObjects.cpp]: Identation of the lines 239 - 245 have to be adjusted.

File: GeoLib/Polygon.cpp [hunk 5]
```diff
@@ -524,31 +523,39 @@ bool operator==(Polygon const& lhs, Polygon const& rhs)
...
-            // new start point at second polygon - the point n-1 of a polygon is equal to the
-            // first point of the polygon (for this reason: n-2)
-            j=n-2;
-            for (; j>k-1; j--) {
-                if (lhs.getPointID(n-2+j+k-2) != rhs.getPointID(j)) {
-                    return false;
-                }
+        }
+        // new start point at second polygon - the point n-1 of a polygon is
+        // equal to the
```
Comment (TomFischer) [inline GeoLib/Polygon.cpp]: Maybe the line break is not necessary here.

File: GeoLib/QuadTree.h [hunk 6]
```diff
@@ -294,7 +294,7 @@ template <typename POINT> class QuadTree
 
         if (this->_father->isChild (this, Quadrant::NW))
             return north_neighbor->getChild (Quadrant::SW);
-        else
+
             return north_neighbor->getChild (Quadrant::SE);
```
Comment (TomFischer) [inline GeoLib/QuadTree.h]: Identation at several places in this file seems to be wrong.

File: MeshGeoToolsLib/GeoMapper.cpp [hunk 7]
```diff
@@ -242,12 +242,12 @@ static MeshLib::Element const* findElementContainingPointXY(
...
-            // clean up the copied nodes
-            for (std::size_t k(0); k<elem_2d->getNumberOfNodes(); ++k) {
-                delete elem_2d->getNode(k);
-            }
         }
+        // clean up the copied nodes
+        for (std::size_t k(0); k < elem_2d->getNumberOfNodes(); ++k)
+        {
+            delete elem_2d->getNode(k);
+            }
```
Comment (TomFischer) [inline MeshGeoToolsLib/GeoMapper.cpp]: Identation seems to be wrong.

File: MeshLib/MeshEditing/MeshRevision.cpp [hunk 8]
```diff
@@ -848,9 +849,12 @@ const std::pair<unsigned, unsigned> MeshRevision::lutHexBackNodes(
...
-    else if ((id1==3 && id2==4) || (id1==4 && id2==3)) return 5;
+    if ((id1 == 1 && id2 == 2) || (id1 == 2 && id2 == 1))
+        return 0;
+    if ((id1 == 0 && id2 == 2) || (id1 == 2 && id2 == 0))
+        return 1;
+    if ((id1 == 3 && id2 == 4) || (id1 == 4 && id2 == 3))
+        return 5;
     else if ((id1==4 && id2==5) || (id1==5 && id2==4)) return 3;
     else if ((id1==3 && id2==5) || (id1==5 && id2==3)) return 4;
     else return std::numeric_limits<unsigned>::max();
```
Comment (TomFischer) [inline MeshLib/MeshEditing/MeshRevision.cpp]: Some `else if` left ...

File: Applications/DataExplorer/VtkVis/VtkFilterFactory.cpp (hunk 3 above)
Comment (endJunction) [inline Applications/DataExplorer/VtkVis/VtkFilterFactory.cpp]: Strange, it was detected only partially. Fixed now.

**Comment by endJunction** [comment]
@TomFischer Many thanks for the careful review.

We might want to clang-format everything: some changes were touching the if and the else keywords only, so clang-format was applied only in vicinity of the changes (and missed the indentation in the else part).
If the code would be already formatted, one could reformat the whole file simply. +32C... :sunny:

**Pull request merged**